crates/rspack_core/src/external_module.rs

```diff
@@ -469,21 +468,15 @@ impl Module for ExternalModule {
       optimization_bailouts: vec![],
     };
     // TODO add exports_type for request
-    match self.external_type.as_str() {
+    match self.resolved_external_type.as_str() {
       "this" => build_result.build_info.strict = false,
       "system" => build_result.build_meta.exports_type = BuildMetaExportsType::Namespace,
       "script" | "promise" => build_result.build_meta.has_top_level_await = true,
-      "module" | "import" | "module-import" => match self.get_module_import_type(external_type) {
-        "module" => build_result.build_meta.exports_type = BuildMetaExportsType::Namespace,
-        "import" => {
-          build_result.build_meta.has_top_level_await = true;
-          build_result.build_meta.exports_type = BuildMetaExportsType::Namespace;
-        }
-        r#type => panic!(
```

Comment:
This should just ignore the request, instead of panic, module-import should only care about import and import(), for other dependency type, it's user's job to tell the compiler what external type to use (by config